test/extended/apiserver/resource_quota.go

```diff
@@ -0,0 +1,61 @@
+package apiserver
+
+import (
+	"context"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Serial]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("apiserver-resource-quota")
+
+	g.It("The number of created persistent volume claims can not exceed the limitation", func() {
+		testProject := oc.SetupProject()
+		testResourceQuotaName := "my-resource-quota-" + testProject
+		pvcName := "myclaim-" + testProject
+		clusterAdminKubeClient := oc.AdminKubeClient()
+
+		rq := &corev1.ResourceQuota{
+			ObjectMeta: metav1.ObjectMeta{Name: testResourceQuotaName, Namespace: testProject},
+			Spec: corev1.ResourceQuotaSpec{
+				Hard: corev1.ResourceList{
+					"persistentvolumeclaims": resource.MustParse("1"),
+				},
+			},
+		}
+
+		pvc := &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pvcName,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("3Gi"),
+					},
+				},
+			},
+		}
+		_, err := clusterAdminKubeClient.CoreV1().ResourceQuotas(testProject).Create(context.Background(), rq, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = clusterAdminKubeClient.CoreV1().PersistentVolumeClaims(testProject).Create(context.Background(), pvc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = clusterAdminKubeClient.CoreV1().PersistentVolumeClaims(testProject).Create(context.Background(), pvc, metav1.CreateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.MatchRegexp(pvcName + `.*forbidden.*[Ee]xceeded quota`))
```

Comment:
Similarly to that other test (https://github.com/openshift/origin/tree/master/test/extended/quota/resourcequota.go) I'd also add a deletion of the PVC and checking if the number in the RQ status went down.

Reply:
Updated